pkg/api/defaults.go

```diff
@@ -266,8 +266,11 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpdate bool) {
 			a.OrchestratorProfile.KubernetesConfig.ProxyMode = DefaultKubeProxyMode
 		}
 
-		// Configure addons
+		// First, Configure addons
```

Comment:
Added some commentary here to emphasize that the current implementation of defaults enforcement depends heavily upon the addons data, and so we must continue to construct our default addons configuration as a pre-requisite for k8s config defaults